e2e2/test/cases/neuron/main_test.go

```diff
@@ -109,28 +108,58 @@ func checkNodeTypes(ctx context.Context, config *envconf.Config) (context.Contex
 		return ctx, fmt.Errorf("Node types are not the same, all node types must be the same in the cluster")
 	}
 
-	if *nodeType != "" {
-		for _, v := range nodes.Items {
-			if v.Labels["node.kubernetes.io/instance-type"] == *nodeType {
-				nodeCount++
-				neuron := v.Status.Capacity["aws.amazon.com/neuron"]
-				neuronPerNode = int(neuron.Value())
-				efa := v.Status.Capacity["vpc.amazonaws.com/efa"]
-				efaPerNode = int(efa.Value())
-			}
+	// Calculate capacities for all nodes
+	nodeCount = 0
+	totalNeuronCount := 0
+	totalNeuronCoreCount := 0
+	totalEfaCount := 0
+
+	for _, node := range nodes.Items {
+		nodeCount++
+		log.Printf("[INFO] Processing node %s", node.Name)
+
+		// Check for Neuron capacity
+		neuron, ok := node.Status.Capacity["aws.amazon.com/neuron"]
+		if ok {
+			totalNeuronCount += int(neuron.Value())
+		} else {
+			log.Printf("[WARN] Node %s does not have 'aws.amazon.com/neuron' capacity", node.Name)
+		}
+
+		// Check for NeuronCore capacity
+		neuronCore, ok := node.Status.Capacity["aws.amazon.com/neuroncore"]
+		if ok {
+			totalNeuronCoreCount += int(neuronCore.Value())
+		} else {
+			log.Printf("[WARN] Node %s does not have 'aws.amazon.com/neuroncore' capacity", node.Name)
 		}
+
+		// Check for EFA capacity
+		efa, ok := node.Status.Capacity["vpc.amazonaws.com/efa"]
+		if ok {
+			totalEfaCount += int(efa.Value())
+		} else {
+			log.Printf("[WARN] Node %s does not have 'vpc.amazonaws.com/efa' capacity", node.Name)
+		}
+	}
+
+	// Calculate per-node capacities
+	if nodeCount > 0 {
```

Comment:
didn't need to construct `nodeCount` above
```suggestion
	if nodeCount := len(nodes.Items); nodeCount > 0 {
```